test/unit/plugins/validate-semantic/2and3/schemas.js

```diff
@@ -169,3 +169,347 @@ describe(`validation plugin - semantic - 2and3 schemas`, () => {
+    })
+  })
+
+  //Array Tests
+
+  it("should return an error for a non array value in a enum array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "arraySample",
+                in: "query",
+                schema: {
+                  type: "array",
+                  enum: [1, 2, 3],     
+                  items: {
+                    type: "array",
+                    items: {
+                      type: "number"
+                    }             
+                  }
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS().filter((err => err.source !== "")) 
+      expect(allErrors.length).toEqual(3)
+      expect(allErrors[0]).toInclude({
+        level: "warning",
+        message: "enum value should conform to its schema's `type`",
+        path: ["paths", "/pets", "get", "parameters", "0", "schema", "enum", 0]
+      })
+      expect(allErrors[1]).toInclude({
+        level: "warning",
+        message: "enum value should conform to its schema's `type`",
+        path: ["paths", "/pets", "get", "parameters", "0", "schema", "enum", 1]
+      })
+      expect(allErrors[2]).toInclude({
+        level: "warning",
+        message: "enum value should conform to its schema's `type`",
+        path: ["paths", "/pets", "get", "parameters", "0", "schema", "enum", 2]
+      })
+    })
+  })
+  
+  it("should not return a type error when all items are array in a enum array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "arraySample",
+                in: "query",
+                schema: {
+                  type: "array",
+                  enum: [[1,2],[3,4]],     
+                  items: {
+                    type: "number",
+                    items: {
+                      type: "number"
+                    }                         
+                  }
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      expect(allErrors.length).toEqual(0)
+    })
+  })
+
+  //Object Tests
+
+  it("should return an error for a non object value (array) in a enum object type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "objectSample",
+                in: "query",
+                schema: {
+                  type: "object",
+                  enum: [[1,3], 2, 3]
```

Comment:
**Change request**: the validator, and this test, should be updated to expect the first enum member to be incorrect as well.

In JSON Schema, arrays should not be considered objects (even though, yes, JavaScript will tell you `typeof []` is `object`).

Reply:
Fixed.